### PR TITLE
MPTF-65

### PR DESCRIPTION
**Pull request by rkrule**
Per Payments Team Request:

-Add PerformReauthorization to ConfirmFunds Request
-Add ReauthorizationAttempted to ConfirmFunds Reply

### CODE REVIEW

File: src/eBayEnterprise/RetailOrderManagement/Payload/Payment/ConfirmFundsRequest.php [hunk 1]
```diff
@@ -80,7 +84,7 @@ protected function serializeContents()
         $paymentContext = $this->getCardNumber() ?
             $this->serializePaymentContext() :
             $this->serializePaymentContextBase();
-        return $paymentContext . $this->serializeAmount();
+        return $paymentContext . $this->serializeAmount() . "<PerformReauthorization>{$this->xmlEncode($this->getPerformReauthorization())</PerformReauthorization>";
```
Comment (ryaan-anthony) [inline src/eBayEnterprise/RetailOrderManagement/Payload/Payment/ConfirmFundsRequest.php]: to keep this clean you should return this string value from a protected `serializePerformReauthorization` method

File: src/eBayEnterprise/RetailOrderManagement/Payload/Payment/ConfirmFundsReply.php [hunk 2]
```diff
@@ -113,7 +118,7 @@ protected function serializeContents()
             $this->serializePaymentContext() :
             $this->serializePaymentContextBase();
         return $paymentContext .
-        "<FundsAvailable>{$this->xmlEncode($this->getFundsAvailable())}</FundsAvailable>";
+        "<FundsAvailable>{$this->xmlEncode($this->getFundsAvailable())}</FundsAvailable>" . "<ReauthorizationAttempted>{$this->xmlEncode($this->getReauthorizationAttempted())</ReauthorizationAttempted>";
```
Comment (ryaan-anthony) [inline src/eBayEnterprise/RetailOrderManagement/Payload/Payment/ConfirmFundsReply.php]: to keep this clean you should return this string value from a protected `serializeReauthorizationAttempted` method

File: src/eBayEnterprise/RetailOrderManagement/Payload/Payment/ConfirmFundsReply.php [hunk 3]
```diff
@@ -144,4 +149,15 @@ protected function getXmlNamespace()
...
     }
+
+    public function getReauthorizationAttempted()
+    {
+	return $this->reauthorizationAttempted;
+    }
+
+    public function setReauthorizationAttempted($reauthorizationAttempted)
+    {
+	$this->reauthorizationAttempted = $reauthorizationAttempted;
```
Comment (ryaan-anthony) [inline src/eBayEnterprise/RetailOrderManagement/Payload/Payment/ConfirmFundsReply.php]: should this be cast as `(bool)`?

File: src/eBayEnterprise/RetailOrderManagement/Payload/Payment/ConfirmFundsRequest.php [hunk 4]
```diff
@@ -180,4 +184,15 @@ public function setCurrencyCode($code)
...
     }
+
+    public function getPerformReauthorization()
+    {
+	return $this->performReauthorization;
+    }
+
+    public function setPerformReauthorization($performReauthorization)
+    {
+	$this->performReauthorization = $performReauthorization;
```
Comment (ryaan-anthony) [inline src/eBayEnterprise/RetailOrderManagement/Payload/Payment/ConfirmFundsRequest.php]: should this be cast as `(bool)`?

**Comment by ryaan-anthony** [comment]
You should declare these new methods for the validator: https://github.com/RadialCorp/RetailOrderManagement-SDK/blob/master/src/eBayEnterprise/RetailOrderManagement/Payload/PayloadConfigMap.php#L73

**Comment by rkrule** [comment]
Added protected serialize function for PerformReauth and ReauthAttempted.
Added validator for optional groups for the above properties per ConfirmFunds Request / Reply.
Casted setter property for the above to a boolean.

**Comment by ryaan-anthony** [comment]
LGTM

**Pull request merged**